storage/src/main/java/tech/pegasys/teku/storage/client/CombinedChainDataClient.java

```diff
@@ -269,6 +269,34 @@ public SafeFuture<Optional<BeaconState>> getLatestStateAtSlot(final UInt64 slot)
     return historicalChainData.getLatestFinalizedStateAtSlot(slot);
   }
 
+  /**
+   * Returns the latest available state prior to the given slot on the current chain.
+   *
+   * @param slot the latest slot to get the state for
+   * @return the State at or before slot
+   */
+  public SafeFuture<Optional<BeaconState>> getLatestAvailableFinalizedState(final UInt64 slot) {
+    if (!isChainDataFullyAvailable()) {
+      return STATE_NOT_AVAILABLE;
+    }
+
+    if (isRecentData(slot)) {
+      return recentChainData
+          .retrieveStateInEffectAtSlot(slot)
+          .thenCompose(
+              recentState -> {
+                if (recentState.isPresent()) {
+                  return completedFuture(recentState);
+                }
+                // Fall-through to historical query in case state has moved into historical range
+                // during processing
+                return historicalChainData.getLatestAvailableFinalizedState(slot);
+              });
+    }
```

Comment:
Given the method is explicitly asking for a finalized state I think we can just remove this part which is checking for non-finalised states. Just delegate straight to `historicalChainData.getLatestAvailableFinalizedState`